File: apps/ios/ScribbleFit/Sources/Core/Database/Models/WorkoutSet.swift
import Foundation
import SwiftData

@Model public final class WorkoutSet {
    public var id: String
    public var weight: Double
    public var reps: Int
    public var rpe: Double?
    public var notes: String?
    public var exerciseId: String
    public var workout: WorkoutLog?

    public init(id: String, weight: Double, reps: Int, rpe: Double? = nil, notes: String? = nil, exerciseId: String) {
        self.id = id
        self.weight = weight
        self.reps = reps
        self.rpe = rpe
        self.notes = notes
        self.exerciseId = exerciseId
    }
}

File: apps/android/core/database/src/main/java/com/scribblefit/core/database/entity/set/WorkoutSet.kt
package com.scribblefit.core.database.entity.set

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey
import com.scribblefit.core.database.entity.exercise.WorkoutExercise

/**
 * Represents a single set performed for a WorkoutExercise.
 */
@Entity(
    tableName = "workout_set",
    foreignKeys = [
        ForeignKey(
            entity = WorkoutExercise::class,
            parentColumns = ["workoutExerciseId"],
            childColumns = ["workoutExerciseId"],
            onDelete = ForeignKey.CASCADE,
        ),
    ],
    indices = [
        Index(value = ["workoutExerciseId"]),
    ],
)
data class WorkoutSet(
    @PrimaryKey(autoGenerate = true)
    val setId: Long = 0,
    val workoutExerciseId: Long,
    val setNumber: Int,
    val reps: Int,
    val weight: Float,
    val rpe: Float? = null,
    val notes: String? = null,
)
